### DISCUSSION TITLE
Exec not running in order from top down in config file?

### DISCUSSION THREAD

**Discussion by Momijie** [i3/i3#4635] [Q&A] [answered]
On startup, I want to run several terminals and I would prefer to have them in a certain order, such as:
```
╭――――――――――――――――――╮
│       |   top    │
│       |--------- │
│       |          │
╰――――――――――――――――――╯

```
I tried achieving it this way in the i3 config.
```
exec --no-startup-id kitty
exec --no-startup-id kitty -e top
exec --no-startup-id i3-msg 'split v; exec kitty'

I also tried 

exec --no-startup-id i3-msg 'split h; exec kitty; split h; exec kitty -e top;'
exec --no-startup-id i3-msg 'split v; exec kitty;'
```
Even when calling 2 terminals separately they seem to be served whenever which one loads first. Am I misunderstanding how to use exec and maybe even i3-msg?
Is there a way to achieve my desired result?

**Comment by Airblader** [comment, marked as answer]
We do execute them in order (though we don't guarantee that), but we don't wait for them to finish before running the next one. Doing so would obviously be problematic, because it'd mean that your second terminal doesn't open until you close the first one again.

What you have here is a classic race condition. How long it takes from executing some binary until a window opens depends on many things, not the least of which is the application itself (who's to say it opens any windows, or when it does so?)

What you're looking for is [layout saving and restoring](https://i3wm.org/docs/layout-saving.html).
